### PR TITLE
Debug the fuckery

### PR DESCRIPTION
**Pull request by RobinBurri**
Small feats:
- delete some prints
- add checks after send()
- change some values to in config.json for test
- change the order of function call in Request::parse_buffer() 
  => empty_path_handler() after check-path-and_method()
- add try catch in Cluster::setup() to handle socket construction errors
- add a new website (webempty) to test it with siege for eval
- change Cluster::setup() to instantiate new_socket on the heap + delete them if server close

### CODE REVIEW

**Review by 0xCAF3D0OD — DISMISSED** [review]
OK

File: src/cluster/cluster.cpp [hunk 1]
```diff
@@ -36,15 +36,22 @@ Cluster::setup()
...
+			Socket *new_socket = new Socket(AF_INET, it->get("port").get<double>(), SOCK_STREAM, 0, *it);
+			int		fd = new_socket->get_socket_id();
+			FD_SET(fd, &_master_fd_set);
+			_sockets.insert(std::make_pair(fd, new_socket));
+			if (fd > _max_fd)
+				_max_fd = fd;
+			std::cout << it->get("hostname").get<std::string>() << ":" << it->get("port").get<double>()
+					  << std::endl;
+		}
+		catch (std::runtime_error)
```
Comment (t-h2o) [inline src/cluster/cluster.cpp:50]: does that compile with `runtime` ?
```suggestion
		catch (std::exception & e)
```

File: src/cluster/cluster.cpp [hunk 2]
```diff
@@ -60,7 +67,7 @@ Cluster::run()
 		while (select_return == 0)
 		{
 			FD_ZERO(&reading_set);
-			std::memcpy(&reading_set, &_master_fd_set, sizeof(_master_fd_set));
+			FD_COPY(&_master_fd_set, &reading_set);
```
Comment (t-h2o) [inline src/cluster/cluster.cpp:70]: I didn't find this macro in `man 2 select`

**Review by t-h2o — COMMENTED** [review]
:wine_glass:

File: src/cluster/cluster.cpp (hunk 2 above)
Comment (t-h2o) [inline src/cluster/cluster.cpp]: Why ?

Comment (RobinBurri) [reply src/cluster/cluster.cpp]: Well this macro is present in man 2 select on mac
Changed it back
ea11b5cceee7e283c4bda702fec44877850c5a1c

File: src/cluster/cluster.cpp [hunk 3]
```diff
@@ -60,7 +67,7 @@ Cluster::run()
 		while (select_return == 0)
 		{
 			FD_ZERO(&reading_set);
-			std::memcpy(&reading_set, &_master_fd_set, sizeof(_master_fd_set));
+			memcpy(&reading_set, &_master_fd_set, sizeof(_master_fd_set));
```
Comment (t-h2o) [inline src/cluster/cluster.cpp:70]: use `std::memcpy` instead of `memcpy` ?

Comment (RobinBurri) [reply src/cluster/cluster.cpp]: Done
2d9ebbd072fa8e34fc0dc4185e738866473dfc52

**Review by t-h2o — APPROVED** [review]
YOLO

**Review by 0xCAF3D0OD — APPROVED** [review]

**Pull request merged**